lib/marked.js

```diff
@@ -91,28 +97,18 @@ block.normal = merge({}, block);
  */
 
 block.gfm = merge({}, block.normal, {
-  fences: /^ {0,3}(`{3,}|~{3,})([^`\n]*)\n(?:|([\s\S]*?)\n)(?: {0,3}\1[~`]* *(?:\n+|$)|$)/,
-  paragraph: /^/,
-  heading: /^ *(#{1,6}) +([^\n]+?) *#* *(?:\n+|$)/
+  nptable: /^ *([^|\n ].*\|.*)\n *([-:]+ *\|[-| :]*)(?:\n((?:.*[^>\n ].*(?:\n|$))*)\n*|$)/,
```

Comment:
Super-linear.

1. The construct `.*\|.*` can be exploited through a run of `|||...|`. Can we replace with `[^|]*|[^|]*` ?
2. I believe the same problem applies later in the `(?:.*[^>\n ].*` section.